### ISSUE TITLE
supportsInteractiveDescription behaves differently in package.json vs query parameter

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#719]
`supportsInteractiveDescription=true` in package.json behaves differently than the query parameter.  The latter blows right past `hasKeyboardHelpContent: true`.

<details>
<summary>Slack discussion</summary>

Chris Malley  2:33 PM
I’ve added option hasKeyboardHelpContent: true to fourier-making-waves-main.js, thinking that I would get a default keyboard help dialog, ala Coulomb’s Law.  But I don’t see a keyboard icon in the navbar.   When can I find instructions on how add keyboard help content? (edited) 

I’m running with ?supportsInteractiveDescription=true , while Coulomb’s Law has this in package.json:
```js
    "features":{
      "supportsInteractiveDescription": true
    },
```

Jesse Greenberg  2:38 PM
That sounds buggy, checking

Chris Malley  2:39 PM
Yep, sure is buggy. When I add "supportsInteractiveDescription": true to package.json, then I get:
> assert.js:25 Uncaught Error: Assertion failed: screen should have keyboardHelpNode
    at window.assertions.assertFunction (assert.js:25)
    at KeyboardHelpButton.js:78
    at Property.link (Property.js:378)
    at new KeyboardHelpButton (KeyboardHelpButton.js:
2:40
So entirely different behavior based on whether supportsInteractiveDescription: true is in package.json vs query parameter.

Jesse Greenberg  2:40 PM
Right, that isn't correct

</details>

**Comment by pixelzoom** [comment]
Also...  If `supportsInteractiveDescription=true` in package.json, and I try to run with `?supportsInteractiveDescription=false`, it's still trying to create keyboard help dialog content.

**Comment by jessegreenberg** [comment]
The creation of the KeyboardHelpButton is behind a check for `sim.accessibilityPartOfTheAPI` with a comment
> // If the sim has accessibility support in its API, then create the button. This is support consistent API for PhET-iO

`sim.accessibilityPartOfTheAPI` is true only if `supportsInteractiveDescription` is in package.json.

So in order to have a consistent PhET-iO API we are only including the button when accessibility is more "permanent" for the sim. If we didn't care about PhET-iO, this is the kind of behavior I would like

<details>

```patch
Index: js/A11yButtonsHBox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/A11yButtonsHBox.js	(revision 5c8d190f4b9a8c033f9d98504f2e3ae8518efa61)
+++ js/A11yButtonsHBox.js	(date 1622582250131)
@@ -46,8 +46,10 @@
       }
     }
 
-    // If the sim has accessibility support in its API, then create the button. This is support consistent API for PhET-iO
-    if ( sim.accessibilityPartOfTheAPI && sim.hasKeyboardHelpContent ) {
+    // Only show the keyboard help button if the sim supports Interactive Description, there is keyboard help content,
+    // and we are not in mobile safari. phet.chipper.queryParameters.supportsInteractiveDescription defaults to
+    // the fields set by package.json so it is inclusive of all the ways Interactive Description can be enabled.
+    if ( phet.chipper.queryParameters.supportsInteractiveDescription && sim.hasKeyboardHelpContent && !platform.mobileSafari ) {
 
       // Pops open a dialog with information about keyboard navigation
       const keyboardHelpButton = new KeyboardHelpButton(
@@ -55,12 +57,7 @@
         backgroundColorProperty,
         tandem.createTandem( 'keyboardHelpButton' )
       );
-
-      // only show the keyboard help button if the sim supports interactive description, there is keyboard help content,
-      // and we are not in mobile safari
-      if ( phet.chipper.queryParameters.supportsInteractiveDescription && !platform.mobileSafari ) {
-        a11yButtons.push( keyboardHelpButton );
-      }
+      a11yButtons.push( keyboardHelpButton );
     }
 
     assert && assert( !options.children, 'A11yButtonsHBox sets children' );
```
</details>

For the PhET-iO team, is there a way we can accomplish this while also keeping a consistent PhET-iO API?

**Comment by pixelzoom** [comment]
To clarify... That comment:

```js
// If the sim has accessibility support in its API, then create the button. This is support consistent API for PhET-iO
```

.... is in A11yButtonsHBox.js.  And it really should be revised, because the second sentence makes no sense to me.

**Comment by jessegreenberg** [comment]
Would this work? In `phet` brand, the button is included as long as Interactive Description is enabeld. In `phet-io` brand, it is included only when accessibility is enabled from package.json.

```patch
Index: js/A11yButtonsHBox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/A11yButtonsHBox.js	(revision 5c8d190f4b9a8c033f9d98504f2e3ae8518efa61)
+++ js/A11yButtonsHBox.js	(date 1622582970897)
@@ -47,7 +47,9 @@
     }
 
     // If the sim has accessibility support in its API, then create the button. This is support consistent API for PhET-iO
-    if ( sim.accessibilityPartOfTheAPI && sim.hasKeyboardHelpContent ) {
+    if ( sim.hasKeyboardHelpContent &&
+         ( phet.chipper.brand === 'phet' && phet.chipper.queryParameters.supportsInteractiveDescription ) ||
+         ( phet.chipper.brand === 'phet-io' && sim.accessibilityPartOfTheAPI ) ) {
 
       // Pops open a dialog with information about keyboard navigation
       const keyboardHelpButton = new KeyboardHelpButton(
```

**Comment by samreid** [comment]
The proposed patch looks like it will do what we want.  For PhET-iO, we should create the button if it is supported, even if a query parameter may choose not to show it.  This is to support a consistent API for PhET-iO.  What should we do for other brands, such as adapted-from-phet, or client-built brands?  Would something like this work (untested)?  Note the patch also improves upon the broken comment.

```diff
Index: js/A11yButtonsHBox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/A11yButtonsHBox.js b/js/A11yButtonsHBox.js
--- a/js/A11yButtonsHBox.js	(revision 5c8d190f4b9a8c033f9d98504f2e3ae8518efa61)
+++ b/js/A11yButtonsHBox.js	(date 1622603007184)
@@ -46,8 +46,9 @@
       }
     }
 
-    // If the sim has accessibility support in its API, then create the button. This is support consistent API for PhET-iO
-    if ( sim.accessibilityPartOfTheAPI && sim.hasKeyboardHelpContent ) {
+    // If the sim has accessibility support in its API, create the button. This is to support a consistent PhET-iO API
+    if ( sim.hasKeyboardHelpContent && ( phet.chipper.brand === 'phet-io' ? sim.accessibilityPartOfTheAPI :
+                                         phet.chipper.queryParameters.supportsInteractiveDescription ) ) {
 
       // Pops open a dialog with information about keyboard navigation
       const keyboardHelpButton = new KeyboardHelpButton(
```

**Comment by jessegreenberg** [comment]
OK, thanks! That looks better, you are right it should be available phet, adapted, or any non phet-io brand. Ill proceed with that change.

**Comment by jessegreenberg** [comment]
I added this and verified that for phet and adapted-from-phet brands the KeyboardHelpButton is added whenever the sim  has Interactive Description enabled, and the query parameter overrides the package.json entry. For phet-io brand, it only is available if `supportsInteractiveDescription` is true in package.json, and is not included when enabled with query parameter.

@pixelzoom would you like to review or test that this is working as you expect now?

**Comment by pixelzoom** [comment]
I did a couple of quick tests with Fourier:

* `supportsInteractiveDescription: true` in package.json, `?supportsInteractiveDescription=false`
* No `supportsInteractiveDescription` in package.json, `?supportsInteractiveDescription=true`

... and they worked as expected. Thanks!
